HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiDevicePath.java

```diff
@@ -0,0 +1,441 @@
+  tmpType += HexUtils.byteArrayToHexString(nvmData);
+  subType = tmpType;
+  return tmpType;
+}
+
+/**
+ * BIOS Device Type definition.
+ * From Appendix A of the BIOS Boot Specification.
+ * Only process the Device type.
+ * Status bootHandler pointer, and description String pointer are ignored.
+ * @param path byte array holding the device path.
+ * @return String that represents the UEFI defined BIOS Device Type.
+ */
+private String biosDevicePath(final byte[] path, final int offset) {
+  String devPath = "Legacy BIOS : Type = ";
+  byte devPathType = path[offset + 1];
+  Byte pathType = Byte.valueOf(devPathType);
+  switch (pathType.intValue()) {
+    case UefiConstants.DEVPATH_BIOS_RESERVED: devPath += "Reserved"; break;
+    case UefiConstants.DEVPATH_BIOS_FLOPPY:   devPath += "Floppy"; break;
+    case UefiConstants.DEVPATH_BIOS_HD:       devPath += "Hard Disk"; break;
+    case UefiConstants.DEVPATH_BIOS_CD:       devPath += "CD-ROM"; break;
+    case UefiConstants.DEVPATH_BIOS_PCM:      devPath += "PCMCIA"; break;
+    case UefiConstants.DEVPATH_BIOS_USB:      devPath += "USB"; break;
+    case UefiConstants.DEVPATH_BIOS_EN:       devPath += "Embedded network"; break;
+    case UefiConstants.DEVPATH_BIOS_BEV:      devPath +=
+                                 "Bootstrap Entry Vector (BEV) from an Option ROM";
+    break;
+  default: devPath += "Reserved";
+  break;
+  }
+  subType = devPath;
+  return devPath;
+}
+
+/**
+ * Returns PIWG firmware volume info.
+ * UEFI Specification, Version 2.8.
+ * PIWG Firmware File Section 10.3.5.6:
+ * Contents are defined in the UEFI PI Specification.
+ * @param path
+ * @param offset
+ * @return String that represents the PIWG Firmware Volume Path
+  */
+private String piwgFirmVolFile(final byte[] path, final int offset) {
+  String fWPath = "PIWG Firmware File ";
+  byte[] guidData = new byte[UefiConstants.SIZE_16];
+  System.arraycopy(path, UefiConstants.OFFSET_4 + offset, guidData, 0, UefiConstants.SIZE_16);
+  UefiGuid guid = new UefiGuid(guidData);
+  fWPath += guid.toString();
+  subType = fWPath;
+  return fWPath;
+}
+
+/**
+ * Returns PIWG firmware file info.
+ * UEFI Specification, Version 2.8.
+ * PIWG Firmware Volume Section 10.3.5.7:
+ * Contents are defined in the UEFI PI Specification.
+ * @param path
+ * @param offset
+ * @return String that represents the PIWG Firmware Volume Path
+ */
+private String piwgFirmVolPath(final byte[] path, final int offset) {
+  String fWPath = "PIWG Firmware Volume ";
+  byte[] guidData = new byte[UefiConstants.SIZE_16];
+  System.arraycopy(path, UefiConstants.OFFSET_4 + offset, guidData, 0, UefiConstants.SIZE_16);
+  UefiGuid guid = new UefiGuid(guidData);
+  fWPath += guid.toString();
+  subType = fWPath;
+  return fWPath;
+}
+
+/**
+ * Returns a string that represents the UEFi Device path.
+ * @return UEFi Device path.
+ */
+public String toString() {
+  return (devPathInfo);
```

Comment:
why is this surrounded in parentheses?

Reply:
Checkstyles doesn't seem to mind. OK removed.